cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -214,11 +207,17 @@ function(em_add_tracked_link_flag target flagname)
 			# This approach was recommended by http://www.cmake.org/pipermail/cmake/2010-May/037206.html
 			add_custom_command(OUTPUT ${dummy_c_name} COMMAND ${CMAKE_COMMAND} -E touch ${dummy_c_name} DEPENDS ${jsfile})
 			target_link_libraries(${target} ${dummy_lib_name})
+			
+			# Link the js-library to the target
+			# When a linked library starts with a "-" cmake will just add it to the linker command line as it is.
+			# The advantage of doing it this way is that the js-library will also be automatically linked to targets
+			# that depend on this target.			
+			get_filename_component(js_file_absolute_path "${jsfile}" ABSOLUTE )			
```

Comment:
This will make sure that relative paths for js files are converted to absolute path to make sure they work in other projects residing in a different path.